src/main/java/seedu/address/gamemanager/GameManager.java

```diff
@@ -0,0 +1,130 @@
+package seedu.address.gamemanager;
+
+import java.nio.file.Path;
+
+import javafx.application.Platform;
+import javafx.collections.ObservableList;
+
+import seedu.address.commons.core.GuiSettings;
+
+import seedu.address.logic.Logic;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+import seedu.address.model.card.Card;
+
+/**
+ * Class that wraps around the entire apps logic and the GameTimer. This is done to separate all logic
+ * of the game away from the GameTimer entirely, and to separate all GameTimer from the UI itself.
+ */
+public class GameManager {
+
+    private Logic logic;
+    private GameTimer gameTimer = null;
+
+    private TimerDisplayCallBack timerDisplayCallBack = null;
+
+    // Call-back method to update ResultDisplay in MainWindow
+    private ResultDisplayCallBack resultDisplayCallBack = null; // not used for now.
+
+    private MainWindowExecuteCallBack mainWindowExecuteCallBack = null;
+
+    public GameManager(Logic logic) {
+        this.logic = logic;
```

Comment:
So game manager is a higher abstraction level than logic manager?
Looks good.

Reply:
Yep, it will handle both Game commands and other commands. Will rename it accordingly to avoid ambiguity (ie. "AppManager").